xarray/core/indexing.py

```diff
@@ -815,6 +820,17 @@ def __setitem__(self, key, value):
         array, key = self._indexing_array_and_key(key)
         array[key] = value
 
+    def rolling_window(self, axis, window):
```

Comment:
Can we put this in `duck_array_ops` instead? I'd like to keep the indexing adapters simple.